### ISSUE TITLE
Too many redirects after enabling HTTPS

### ISSUE THREAD

**Issue by brylie** [RocketChat/Rocket.Chat#2216]
On a new installation, with a self-signed certificate, we get the following error(s) after enabling the 'force SSH' setting:

Firefox:
```
Firefox has detected that the server is redirecting the request for this address in a way that will never complete.
```

Chrome:
```
The webpage at https://chat. ... .com/ has resulted in too many redirects.
```

How can we fix this issue?

**Comment by engelgabriel** [comment]
What version are you using?
Did you update the Site_URL on the admin panel?

**Comment by shantanuthatte** [comment]
Add this to your nginx config:
```
      proxy_set_header X-Forwarded-Proto https;
```
FYI, here's my nginx config:
```
upstream chat_server {
    server 127.0.0.1:6000;
    keepalive 8;
}

# the nginx server instance
server {
    listen         80;
    server_name    chat.server.com;
    return         301 https://$server_name$request_uri;
}
server {
    server_name chat.server.com;
    access_log /var/log/nginx/chat.server.log;
    
    #Start SSL config
    listen 443 ssl;
    add_header Strict-Transport-Security "max-age=31536000"; 
    ssl_certificate /path/to/certificate.pem;
    ssl_certificate_key /path/to/privkey.pem;
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_prefer_server_ciphers on;
    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';

    # pass the request to the node.js server with the correct headers
    # and much more can be added, see nginx config options
    location / {
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto https;
      proxy_set_header Host $http_host;
      proxy_set_header X-NginX-Proxy true;

      proxy_pass http://chat_server/;
      proxy_redirect off;
    }
    location /sockjs/ {
        proxy_pass http://chat_server;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }
}
```

**Comment by engelgabriel** [comment]
Please make sure you are using Rocket.Chat version above 0.17.0

You can check this by opening your version of the URL https://demo.rocket.chat/api/info

Also, make sure you update the Site URL and add the X-Forwarded-Proto to your reverse proxy.

**Comment by brylie** [comment]
OK, I installed Rocket.Chat via Ansible. Where can I find the nginx.conf? I have located one config file at `/etc/nginx/nginx.conf`, but it does not seem similar to the one previously posted.

**Comment by brylie** [comment]
OK, we found the rocket_chat.conf, and added the suggested lines. That seems to work. Now, we have a 502 Bad Gateway! :-)

**Comment by brylie** [comment]
Also, part of the problem was that our hostname was not properly defined in /etc/hosts, it was just `chat` without a top level domain. The `server_name` value was set to `chat`, as opposed to `chat.example.com`.

**Comment by brylie** [comment]
Previously to resolve the **Bad Gateway**, I have re-run the ansible playbook. However, running the Ansible playbook seems to reset the `rocket_chat.conf`, so that it no longer contains the ` proxy_set_header X-Forwarded-Proto https;` line.

**Comment by engelgabriel** [comment]
Ok, I've added it now
https://github.com/RocketChat/Rocket.Chat.Ansible/commit/32483c1c906f575f6d5c2a880f03148d5a56601e?ts=2

**Comment by brylie** [comment]
Thanks @engelgabriel. I will try to reinstall. This is still somewhat confusing, and we get a `502 Bad Gateway` as soon as we initially installed, as well as redirects to `https://chat` instead of `https://chat.domain.com` when enabling HTTPS.

**Comment by engelgabriel** [comment]
If you are using NGINX, you should not need to set the FORCE SSL, as it would be done by the NGINX anyway. The setting only applies when you dont have nginx.

**Comment by brylie** [comment]
OK, how can I disable the setting? I am unable to access the site.

**Comment by brylie** [comment]
This may be the reason for the redirect loop.

**Comment by brylie** [comment]
Also, it might be good to somehow disable that setting in the Admin UI on nginx instances, if it might potentially disable a site.

**Comment by daviddesmedt** [comment]
@brylie : have you managed to find a way to disable this setting? My UI is gone as well.

**Comment by daviddesmedt** [comment]
@engelgabriel : can you tell me how I can disable this setting? I have used to UI to FORCE SLL as well, resulting in a un accessable website. Can I disable this setting on my server or config file?

**Comment by engelgabriel** [comment]
You need to go into MongoDB and delete the object with `_id: "Force_SSL"` form the `rocketchat_settings` collection.

**Comment by daviddesmedt** [comment]
Thanks a lot. Will try this.

Verzonden vanuit Outlook Mobile<https://aka.ms/blhgte>



On Tue, Mar 8, 2016 at 8:18 AM -0800, "Gabriel Engel" <notifications@github.com<mailto:notifications@github.com>> wrote:


You need to go into MongoDB and delete the object with _id: "Force_SSL" form the rocketchat_settings collection.

-

**Comment by shtefcs** [comment]
I have this problem as well. It's started happening week a go, and now more and more users reporting problem that they can't enter the chat.

My version is 0.50.1

I have not **Force SSL** turned on

**Comment by blamoo** [comment]
I had the same problem today, and found the solution for my case.

I was [changing the port number](https://rocket.chat/docs/installation/manual-installation/ubuntu/snaps/#how-do-i-change-rocketchat-port-mongo_url-and-mongo_oplog_url-in-my-snap) of the main service to 80 after the installation.
But I had the redirection loop problem when I enabled SSL with caddy because it tried to use port 80 too.
I fixed it by reinstalling everything and leaving the default port 3000 at the main service. Now both ports 80 and 443 are being proxied by Caddy to the internal port 3000, which is hid behind a firewall.
